frontend/src/components/modals/RelationsModal.tsx

```diff
@@ -126,37 +126,27 @@ const relationTable: (
           </h3>
         </div>
         <p className={classes(css.subTitle)}>{subTitle}</p>
-        {!tasks.length ? (
-          <div className={classes(css.noRelations)}>
-            <Trans i18nKey="No relations" />
-          </div>
-        ) : (
-          <div
-            className={classes(css.relationList)}
-            style={{ maxHeight: height }}
-          >
-            {tasks.map((t) => (
-              <div key={t.id} className={classes(css.relationRow)}>
-                <TaskRow
-                  style={{
-                    ['--background-color' as any]: colors.black5,
-                    marginLeft: 0,
-                  }}
-                  task={t}
-                  largeIcons
-                  onClick={onTaskClick}
-                  showMilestoneName={showMilestoneNames}
-                />
-                <div className={classes(css.relationAlert)}>
-                  <BadRelationWarning
-                    relation={{ to: t.id, type }}
-                    badRelations={badRelations}
-                  />
-                </div>
-              </div>
-            ))}
-          </div>
-        )}
+        <TaskRelationTable
+          height={height}
+          tasks={tasks}
+          taskProps={{
+            style: {
+              ['--background-color' as any]: colors.black5,
+              marginLeft: 0,
+            },
+            largeIcons: true,
+            onClick: onTaskClick,
+            showMilestoneName: showMilestoneNames,
+          }}
+          Action={({ task: { id } }) => (
+            <div className={classes(css.relationAlert)}>
+              <BadRelationWarning
+                relation={{ to: id, type }}
+                badRelations={badRelations}
+              />
+            </div>
+          )}
```

Comment:
This is highlighted with a react/no-unstable-nested-components -warning. The component could be memoized

`const BadRelation = useCallback(
      ({ task: { id } }) => (
        <div className={classes(css.relationAlert)}>
          <BadRelationWarning
            relation={{ to: id, type }}
            badRelations={badRelations}
          />
        </div>
      ),
      [badRelations],
    );`